src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,103 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private static final double RESIZE_NUMBER = 1.5;
+    private int size;
+    private Object[] elementsData = new Object[DEFAULT_SIZE];
+
+    public void resizeLength() {
+        if (size == elementsData.length) {
+            Object[] biggerArray = new Object[(int) (elementsData.length * RESIZE_NUMBER)];
+            System.arraycopy(elementsData, 0, biggerArray, 0, size);
+            elementsData = biggerArray;
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        resizeLength();
+        elementsData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        resizeLength();
+        if (index > size || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index isn't correct");
+        } else {
+            System.arraycopy(elementsData, index, elementsData, index + 1, size - index);
+            elementsData[index] = value;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        Object[] compliet = new Object[size + list.size()];
+        System.arraycopy(elementsData, 0, compliet, 0, size);
+        for (int i = 0; i < list.size(); i++) {
+            compliet[size + i] = list.get(i);
+        }
+        size = compliet.length;
+        elementsData = compliet;
```

Comment:
call the method `add`
```suggestion
        for (int i = 0; i < list.size(); i++) {
            add(list.get(i));
        }
```

Reply:
+

Reply:
sorry i miss this(